rpc/core/tx.go

```diff
@@ -68,47 +65,28 @@ func (env *Environment) TxSearch(
 		return nil, err
 	}
 
-	results, err := env.TxIndexer.Search(ctx.Context(), q)
-	if err != nil {
-		return nil, err
+	// Validate number of results per page
+	perPage := env.validatePerPage(perPagePtr)
+	if pagePtr == nil {
+		// Default to page 1 if not specified
+		pagePtr = new(int)
+		*pagePtr = 1
 	}
 
-	// sort results (must be done before pagination)
-	switch orderBy {
-	case "desc":
-		sort.Slice(results, func(i, j int) bool {
-			if results[i].Height == results[j].Height {
-				return results[i].Index > results[j].Index
-			}
-			return results[i].Height > results[j].Height
-		})
-	case "asc", "":
-		sort.Slice(results, func(i, j int) bool {
-			if results[i].Height == results[j].Height {
-				return results[i].Index < results[j].Index
-			}
-			return results[i].Height < results[j].Height
-		})
-	default:
-		return nil, ErrInvalidOrderBy{orderBy}
+	pagSettings := ctypes.Pagination{
+		OrderDesc:   orderBy == "desc",
```

Comment:
So we assume that everything that is not `desc` is `asc`.

Reply:
Yeah, thought it would be better than to just add more logic gates to error on an invalid orderBy, but happy to change if desired (sort of like how we order asc in the previous code if nothing is provided)

Reply:
Added orderBy check, which mirrors the case logic that was previously in place https://github.com/cometbft/cometbft/pull/2855/commits/b7946c4bb7a4ea40f59b3d956a1ee79e58849f92